torch/distributed/tensor/experimental/_attention.py

```diff
@@ -225,8 +226,15 @@ def _scaled_dot_product_ring_efficient_attention(
 ) -> tuple[torch.Tensor, ...]:
     if attn_bias is not None:
         raise NotImplementedError("attn_bias is not supported yet")
+
     if not compute_log_sumexp:
-        raise NotImplementedError("compute_log_sumexp must be set")
+        warning_once(
+            logger,
+            "_scaled_dot_product_ring_efficient_attention requires compute_log_sumexp "
+            f"be True but received {compute_log_sumexp}. "
+            "Override compute_log_sumexp to True.",
+        )
+        compute_log_sumexp = True
```

Comment:
Any reason that it is a parameter to the function if it is always going to be `True`? 

Sorry, I don't have a lot of context on how this method is used, do you have pointers?

Reply:
This function is used in dispatcher when the sdp efficient attention ATEN op is called. So the function signature better stays consistent with the op (i.e. keep the compute_log_sumexp argument even if we require it be True):  https://github.com/pytorch/pytorch/blob/2f317bbdbc416086c907941593b97c947bbbf5d9/torch/distributed/tensor/experimental/_attention.py#L545-L549

Besides, I haven't looked into the reason why `compute_log_sumexp=False` gives error and hopefully we can make it compatible. But the bottom line is, setting `compute_log_sumexp=True` shouldn't have significant overhead. A possible exception is in inference. I'll leave a TODO here and mark this PR as a temporary fix to unblock torchtitan 1.0 release.

Reply:
TBH I dont even think you need to warn here. There is no real way for users to control this and its an optimization to save some memory + some writes for the kernel when backwards wont be used. 

Neither of these are true for the CP use case since you will always need the LSE to merge. So I think just set it to true and dont worry